### PR TITLE
Fix compressed multiindex for output of groupby.rolling

### LINKED ISSUE #36018: QST: is the new behavior of GroupByRolling for MultiIndex in v1.1.1 intended?
#### Question about pandas

Let's say we have a `Series` with `MultiIndex` like the below.

```python
>>> pser = pd.Series(
...     [1, 2, 3, 2],
...     index=pd.MultiIndex.from_tuples([("a", "x"), ("a", "y"), ("b", "z"), ("c", "z")]),
...     name="a",
... )

>>> pser
a  x    1
   y    2
b  z    3
c  z    2
Name: a, dtype: int64
```

Then, when I use `GroupByRolling`, In the version of pandas <= 1.0.5 shows result as below.
(Let's focus on the level of `MultiIndex` in the examples below)

```python
>>> pser.groupby(pser).rolling(2).max()
a
1  a  x    NaN
2  a  y    NaN
   c  z    2.0
3  b  z    NaN
Name: a, dtype: float64

>>> pser.groupby(pser).rolling(2).max().index
MultiIndex([(1, 'a', 'x'),
            (2, 'a', 'y'),
            (2, 'c', 'z'),
            (3, 'b', 'z')],
           names=['a', None, None])

>>> pser.groupby(pser).rolling(2).max().index.nlevels  # It keeps the index level.
3
```

However, In the pandas 1.1.0, the result seems different from the previous version as below.
(The level of `MultiIndex` is decreased after performing)

```python
>>> pser.groupby(pser).rolling(2).max()
a
1  (a, x)    NaN
2  (a, y)    NaN
   (c, z)    2.0
3  (b, z)    NaN
Name: a, dtype: float64

>>> pser.groupby(pser).rolling(2).max().index
MultiIndex([(1, ('a', 'x')),
            (2, ('a', 'y')),
            (2, ('c', 'z')),
            (3, ('b', 'z'))],
           names=['a', None])

>>> pser.groupby(pser).rolling(2).max().index.nlevels  # index level changed from 3 to 2
2
```

Is it intended behavior in pandas 1.1.1. ??

Thanks :)

### PR DESCRIPTION
**Pull request by phofl**
- [x] closes #36018 
- [x] tests added / passed
- [x] passes `black pandas`
- [x] passes `git diff upstream/master -u -- "*.py" | flake8 --diff`
- [x] whatsnew entry

The number of leves for MultiIndex in input was compressed to one, because the tuples representing the Index rows were not unpacked. Also name was used instead of names -> resulted in missing name for every MultiIndex.

### CODE REVIEW

File: pandas/tests/groupby/test_groupby.py [hunk 1]
```diff
@@ -2134,3 +2134,24 @@ def test_groupby_column_index_name_lost_fill_funcs(func):
     result = getattr(df_grouped, func)().columns
     expected = pd.Index(["a", "b"], name="idx")
     tm.assert_index_equal(result, expected)
+
+
+@pytest.mark.parametrize("func", ["max", "min"])
+def test_groupby_rolling_index_changed(func):
+    ds = pd.Series(
```
Comment (mroeschke) [inline pandas/tests/groupby/test_groupby.py:2141]: Could you add `# GH 36018` as a comment?

**Review by mroeschke — COMMENTED** [review]
One comment otherwise LGTM

File: pandas/tests/groupby/test_groupby.py (hunk 1 above)
Comment (phofl) [reply pandas/tests/groupby/test_groupby.py:2142]: Thank for the review. Forgot the comment. Is added now.

File: pandas/tests/groupby/test_groupby.py [hunk 2]
```diff
@@ -2134,3 +2134,25 @@ def test_groupby_column_index_name_lost_fill_funcs(func):
     result = getattr(df_grouped, func)().columns
     expected = pd.Index(["a", "b"], name="idx")
     tm.assert_index_equal(result, expected)
+
+
+@pytest.mark.parametrize("func", ["max", "min"])
+def test_groupby_rolling_index_changed(func):
```
Comment (rhshadrach) [inline pandas/tests/groupby/test_groupby.py:2140]: Put tests near other similar tests - in this case the groupby/rolling tests are in window/test_grouper.

**Review by rhshadrach — CHANGES_REQUESTED** [review]
Thanks @phofl - just a minor request.

File: pandas/tests/groupby/test_groupby.py (hunk 2 above)
Comment (phofl) [reply pandas/tests/groupby/test_groupby.py]: Thanks very much. I searched for them yesterday, but it was already late :)

File: pandas/core/window/rolling.py [hunk 3]
```diff
@@ -2211,17 +2211,19 @@ def _apply(
...
+        grouped_index_name = [*grouped_object_index.names]
         groupby_keys = [grouping.name for grouping in self._groupby.grouper._groupings]
         result_index_names = groupby_keys + grouped_index_name
 
         result_index_data = []
         for key, values in self._groupby.grouper.indices.items():
             for value in values:
-                if not is_list_like(key):
-                    data = [key, grouped_object_index[value]]
+                data = [key] if not is_list_like(key) else [*key]
```
Comment (jreback) [inline pandas/core/window/rolling.py:2221]: use `pandas.core.common.maybe_make_list(key)`

File: pandas/core/window/rolling.py [hunk 4]
```diff
@@ -2211,17 +2211,19 @@ def _apply(
...
+                if is_list_like(grouped_object_index[value]):
+                    grouped_object_index_value = [*grouped_object_index[value]]
```
Comment (jreback) [inline pandas/core/window/rolling.py:2223]: can do the same here i think

File: doc/source/whatsnew/v1.2.0.rst [hunk 5]
```diff
@@ -314,6 +314,7 @@ Groupby/resample/rolling
 - Bug when subsetting columns on a :class:`~pandas.core.groupby.DataFrameGroupBy` (e.g. ``df.groupby('a')[['b']])``) would reset the attributes ``axis``, ``dropna``, ``group_keys``, ``level``, ``mutated``, ``sort``, and ``squeeze`` to their default values. (:issue:`9959`)
 - Bug in :meth:`DataFrameGroupby.tshift` failing to raise ``ValueError`` when a frequency cannot be inferred for the index of a group (:issue:`35937`)
 - Bug in :meth:`DataFrame.groupby` does not always maintain column index name for ``any``, ``all``, ``bfill``, ``ffill``, ``shift`` (:issue:`29764`)
+- Bug in :meth:`Series.groupby.rolling` compressed number of levels of :class:`MultiIndex` in input to one (:issue:`36018`)
```
Comment (jreback) [inline doc/source/whatsnew/v1.2.0.rst:317]: would be ok with doing this for 1.1.2 if you want to change the whatsnew note

**Review by jreback — CHANGES_REQUESTED** [review]

File: pandas/core/window/rolling.py (hunk 4 above)
Comment (phofl) [reply pandas/core/window/rolling.py:2223]: Unfortunately maybe_make_list does not convert tuples to lists. I could not find a function in there doing this.

Comment (phofl) [reply pandas/core/window/rolling.py]: Sorry that was stupid, took me a moment to get it. I changed it.

File: pandas/core/window/rolling.py (hunk 3 above)
Comment (phofl) [reply pandas/core/window/rolling.py]: Thanks very much, that's a nice function

File: doc/source/whatsnew/v1.2.0.rst (hunk 5 above)
Comment (phofl) [reply doc/source/whatsnew/v1.2.0.rst]: Yes of course, I moved the whats new entry

**Review by jreback — APPROVED** [review]

**Comment by jreback** [comment]
thanks @phofl very nice, keep em coming!

**Comment by simonjayhawkins** [comment]
@meeseeksdev backport 1.1.x

**Pull request merged**